### PR TITLE
Create money_honey_cr.py

### PR DESCRIPTION
**Pull request by MichalCab**

### CODE REVIEW

File: money_honey_cr.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+from random import randint
+
+from actual_currencies import CurrenciesChanger
+
+
+class MoneyHoney:
+    """
+    I wasn't sure if it's allowed to use external libraries so I didn't use any.
+    Estimated time of coding 4MD.
+    """
```
Comment (MichalCab) [inline money_honey_cr.py:14]: nice that you use doc

File: money_honey_cr.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
+import json
+import sys
+import getopt
+from decimal import Decimal, InvalidOperation
+from random import randint
+
+from actual_currencies import CurrenciesChanger
+
+
+class MoneyHoney:
```
Comment (MichalCab) [inline money_honey_cr.py:10]: `class MoneyHoney(object):`

File: money_honey_cr.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    """
+    I wasn't sure if it's allowed to use external libraries so I didn't use any.
+    Estimated time of coding 4MD.
+    """
+    def __init__(self):
+        pass
```
Comment (MichalCab) [inline money_honey_cr.py:16]: __init__ with pass?

I think you can skip it

no need to declare

File: money_honey_cr.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+    def arguments_handler(self):
+        """
+        Handles the arguments and flow of the program.
+        """
+        try:
+            opts, args = getopt.getopt(sys.argv[1:], "h", ["help", "amount=", "input_currency=", "output_currency=",
```
Comment (MichalCab) [inline money_honey_cr.py:23]: try argparse https://docs.python.org/2/howto/argparse.html

File: money_honey_cr.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+class MoneyHoney:
+    """
+    I wasn't sure if it's allowed to use external libraries so I didn't use any.
+    Estimated time of coding 4MD.
+    """
+    def __init__(self):
+        pass
+
+    def arguments_handler(self):
```
Comment (MichalCab) [inline money_honey_cr.py:18]: `arguments_handler ` hard to read this function

File: money_honey_cr.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                converted_currencies = {}
+                for currency, currency_amount in converted_amounts.items():
+                    converted_currencies[currency] = format(currency_amount, '.4f')
+
+                output_json['input'] = {'amount': format(amount, '.4f'), 'currency': input_currency}
+                output_json['output'] = converted_currencies
+                print("\n" + json.dumps(output_json, indent=4))
+
+    @staticmethod
+    def get_first_value(my_list):
```
Comment (MichalCab) [inline money_honey_cr.py:82]: :3

File: money_honey_cr.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        return str(converted_amount) + " " + output_currency
+
+    def check_currency(self, currency_changer, name, currency, symbol_currencies=None):
+        """
+        Checks if currency is valid and supported.
+        :param currency_changer: Instance of CurrenciesChanger.
+        :param name: Name of the input.
+        :param currency: Currency symbol or letter code.
+        :param symbol_currencies: Array of currency codes for specific symbol.
+        :return: Check function convert_symbol for returns.
```
Comment (MichalCab) [inline money_honey_cr.py:96]: nice doc :))

File: money_honey_cr.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        """
+        if symbol_currencies is not None and currency.upper() in symbol_currencies:
```
Comment (MichalCab) [inline money_honey_cr.py:98]: maybe 

`symbol_currencies is not None `

to just

`symbol_currencies`

File: money_honey_cr.py [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+        """
+        if symbol in list(currency_changer.currency_signs.keys()):
+            currency_codes = currency_changer.currency_signs[symbol]
+            if input_name == "--input_currency" and len(currency_codes) > 1:
+                if input("Would you care to specify? y/n\n").lower() in ['y', 'yes']:
+
+                    print("You can choose from: [", end='')
+                    for i in range(len(currency_codes)):
+                        if i == len(currency_codes) - 1:
+                            print(currency_codes[i] + "]")
```
Comment (MichalCab) [inline money_honey_cr.py:122]: looks dirty:

` + "]")`

File: money_honey_cr.py [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+                                               currency_codes).upper()
+                else:
+                    num_options = len(currency_codes)
+                    return currency_codes[randint(0, (num_options - 1))]
+            else:
+                return currency_codes[0] if input_name == "--input_currency"\
+                                            and len(currency_codes) == 1 else currency_codes
+        else:
+            # it's not symbol
+            return symbol.upper() if input_name == "--input_currency" else [symbol.upper()]
```
Comment (MichalCab) [inline money_honey_cr.py:136]: `else: else: else: else: ` :/ 

hard to read

File: money_honey_cr.py [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+    @staticmethod
+    def usage():
+        print("--amount\n\tRequired. Can contain only number.\n"
+              "--input_currency\n\tRequired. Can contain only three chars or currency symbol.\n"
+              "--output_currency\n\tOptional. Can contain only three chars or currency symbol.\n"
+              "--no_json\n\tOptional. Will change output format from json to custom one.")
+
+
+if __name__ == "__main__":
+    MoneyHoney().arguments_handler()
```
Comment (MichalCab) [inline money_honey_cr.py:147]: as I understand this is main app.. 

this app can just handle args?

No conversion?

expect something like:
```
money_honey = MoneyHoney()
money_honey.arguments_handler()
money_honey.convert()
```

File: money_honey_cr.py (hunk 2 above)
Comment (ZlatyChlapec) [reply money_honey_cr.py:10]: I guess it doesn't hurt to add it there so I added it in commit [#6](https://github.com/ZlatyChlapec/MoneyHoney/commit/f69f0f6253e8de5851e05a24c577f10b1f4c90ae). As I read up it doesn't matter in Python 3. So I guess it's more just like convention?

File: money_honey_cr.py (hunk 3 above)
Comment (ZlatyChlapec) [reply money_honey_cr.py:16]: Not sure why I left it there. Removed in [#6](https://github.com/ZlatyChlapec/MoneyHoney/commit/f69f0f6253e8de5851e05a24c577f10b1f4c90ae).

File: money_honey_cr.py (hunk 8 above)
Comment (ZlatyChlapec) [reply money_honey_cr.py:98]: Nice I didn't know that Python supports checking it in that way. Changed in [#6](https://github.com/ZlatyChlapec/MoneyHoney/commit/f69f0f6253e8de5851e05a24c577f10b1f4c90ae).

File: money_honey_cr.py (hunk 2 above)
Comment (MichalCab) [reply money_honey_cr.py:10]: > more just like convention

exactly @ZlatyChlapec

File: money_honey_cr.py (hunk 4 above)
Comment (ZlatyChlapec) [reply money_honey_cr.py:23]: This thing looks way cleaner. Added in [#7](https://github.com/ZlatyChlapec/MoneyHoney/commit/27c422c0ff89951c23ff16acb3dd78deae4de258).

File: money_honey_cr.py (hunk 5 above)
Comment (ZlatyChlapec) [reply money_honey_cr.py:18]: Yea reworked in [#7](https://github.com/ZlatyChlapec/MoneyHoney/commit/27c422c0ff89951c23ff16acb3dd78deae4de258).

File: money_honey_cr.py (hunk 9 above)
Comment (ZlatyChlapec) [reply money_honey_cr.py:122]: tweaked out this part in [#7](https://github.com/ZlatyChlapec/MoneyHoney/commit/27c422c0ff89951c23ff16acb3dd78deae4de258).

File: money_honey_cr.py (hunk 10 above)
Comment (ZlatyChlapec) [reply money_honey_cr.py:136]: I wasn't able to get completely rid of it but I tried to make it more readable. [#7](https://github.com/ZlatyChlapec/MoneyHoney/commit/27c422c0ff89951c23ff16acb3dd78deae4de258)

File: money_honey_cr.py (hunk 11 above)
Comment (ZlatyChlapec) [reply money_honey_cr.py:147]: Changed to be more like this in [#7](https://github.com/ZlatyChlapec/MoneyHoney/commit/27c422c0ff89951c23ff16acb3dd78deae4de258)

**Pull request closed without merging**